src/main/java/workInMicrosoft/SenderRunnable.java

```diff
@@ -0,0 +1,18 @@
+package workInMicrosoft;
+
+public class SenderRunnable implements Runnable {
+    private int startIndex;
+    private int endIndex;
+
+    public SenderRunnable(int startIndex, int endIndex) {
+        this.startIndex = startIndex;
+        this.endIndex = endIndex;
+    }
+
+    @Override
+    public void run() {
+        for (int i = startIndex; i < endIndex; i++) {
```

Comment:
Последнее письмо не будет обработано

Reply:
```suggestion
        for (int i = startIndex; i < endIndex; i++) {
```
У меня был увеличен диапазон значений до +1, чтобы последним был именно 200*i+1, т.е. 1000 был последним.
Исправил логику
`Thread thread = new Thread(new SenderRunnable((i * batchSize) + 1, ((i + 1) * batchSize)));`
`for (int i = startIndex; i <= endIndex; i++)`